framework/src/modules/chain/synchronizer/utils.js

```diff
@@ -42,6 +45,44 @@ const restoreBlocks = async (blocksModule, processorModule, tx) => {
 	return true;
 };
 
+/**
+ * Allows to restore blocks if there are blocks left upon startup in temp_block table (e.g. node crashed)
+ * Depends upon fork choice rule if blocks will be applied
+ *
+ * 1. Gets all blocks from temp_block table
+ * 2. Sort blocks according to height as we want to lowest height (the next block to be applied)
+ * 3. Uses next block with fork choice rule - if fork status indicates we should switch to different chain
+ * we continue applying blocks using the `restoreBlocks` function.
+ * Otherwise we truncate the temp_block table.
+ *
+ * @param {Object} blocksModule - injection of blocks module object
+ * @param {Object} processorModule - injection of processor module object
+ * @param {Object} storageModule - injection of storage module object
+ * @return {Promise<void>}
+ */
+const restoreBlocksUponStartup = async (
+	blocksModule,
+	processorModule,
+	storageModule,
+) => {
+	// Get all blocks and find lowest height (next one to be applied)
+	const tempBlocks = await storageModule.entities.TempBlock.get();
+	const blockLowestHeight = tempBlocks.reduce((prev, current) =>
+		prev.height < current.height ? prev : current,
+	);
```

Comment:
initial value missing here

Reply:
Do you mean the init value for reducer? This is optional + I check in processor if there are blocks left in the temp_block table, if not, this function is not called.

Reply:
yea it is optional so it will be undefined. so the first run of the `prev` will be `undefined` and u r using `undefined.height`?

Reply:
I verified this, by default your initial value is set to the first object in the array so this won't happen.
+ defining a default value gives more overhead as you never know the hight of the blocks in the array as you have to set a very high initial value (made a mistake first by setting `{ height: 0}`.